### PR TITLE
refactor(platform-browser): specify return type of parseEventName

### PR DESCRIPTION
**Pull request by sonukapoor**
This PR is a followup of the original PR [here](https://github.com/angular/angular/pull/37274).

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by pkozlowski-opensource — APPROVED** [review]
Reviewed previously as https://github.com/angular/angular/pull/37274 and still LGTM

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:322820628:BASE:322820674:1595527001772:75bf10a1)

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:322820628:BASE:322820674:1595527001772:75bf10a1)

**Comment by mhevery** [comment]
Reverted as this caused a breakage in `google3`

**Comment by mhevery** [comment]
[presubmit](http://test/OCL:322820628:BASE:324075442:1596142243356:ebbe9a13)

File: packages/platform-browser/src/dom/events/key_events.ts [hunk 1]
```diff
@@ -136,7 +136,10 @@ export class KeyEventsPlugin extends EventManagerPlugin {
       return null;
     }
 
-    const result: {[k: string]: string} = {};
+    // NOTE: Please don't rewrite this as so, as it will break JSCompiler property renaming.
+    //       The code must remain in the `result['domEventName']` form.
+    // return {domEventName, fullKey};
+    const result: {fullKey: string, domEventName: string} = {} as any;
```
Comment (alfaproject) [inline packages/platform-browser/src/dom/events/key_events.ts:142]: `any` sucks ^_^
```suggestion
    const result = {} as {fullKey: string, domEventName: string};
```

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:325089314:BASE:325089335:1596659367914:69b18257)

**Pull request closed without merging**